app/controllers/cart_controller.rb

```diff
@@ -0,0 +1,17 @@
+class CartController < ApplicationController
+  load_and_authorize_resource class: Order
+
+  def show
+    @order = current_user_order
+  end
+
+  def update
+    coupon = CouponService.new(current_user_order, params[:code])
```

Comment:
Services. It an object which responds to message `call` and returns some thing. Usually it is nil in error case and object for success.
advantages:
- you have always one interface with all your services
- Don't worry about naming. For example:
```ruby
# may be saw smth like
StrangePaginator.new(attrs).paginate
# but better way is:
YourActionService.call(attributes)
```
After refactoring code can looks like this:
```ruby
if ApplyCouponService.call(current_order, params[:code])
  flash[:notice] = I18n.t('success_coupon_use')
else
  flash[:alert] = I18n.t('invalid_coupon')
end

redirect_to cart_path
```
Can you refactor your services?